lang/rust/avro/src/decode.rs

```diff
@@ -68,10 +68,21 @@ fn decode_seq_len<R: Read>(reader: &mut R) -> AvroResult<usize> {
 
 /// Decode a `Value` from avro format given its `Schema`.
 pub fn decode<R: Read>(schema: &Schema, reader: &mut R) -> AvroResult<Value> {
-    fn decode0<R: Read>(
+    let rs = ResolvedSchema::try_from(schema)?;
+    decode_internal(schema, rs.get_names(), &None, reader)
+}
+
+fn decode_internal<R: Read>(
```

Comment:
is there a need of `decode_internal0` now ?

Reply:
you're right there probably is not? as far as I can tell.